core/ipc/rpc/camel-impl/src/main/java/org/opennms/core/rpc/camel/CamelRpcClientPreProcessor.java

```diff
@@ -43,7 +63,7 @@ public void process(Exchange exchange) {
         final JmsQueueNameFactory queueNameFactory = new JmsQueueNameFactory(CamelRpcConstants.JMS_QUEUE_PREFIX,
                 wrapper.getModule().getId(), wrapper.getRequest().getLocation());
         exchange.getIn().setHeader(CamelRpcConstants.JMS_QUEUE_NAME_HEADER, queueNameFactory.getName());
-        exchange.getIn().setHeader(CamelRpcConstants.CAMEL_JMS_REQUEST_TIMEOUT_HEADER, wrapper.getRequest().getTimeToLiveMs());
+        exchange.getIn().setHeader(CamelRpcConstants.CAMEL_JMS_REQUEST_TIMEOUT_HEADER, CAMEL_JMS_REQUEST_TIMEOUT == 0 ? wrapper.getRequest().getTimeToLiveMs() : CAMEL_JMS_REQUEST_TIMEOUT);
```

Comment:
If the request specifies a timeout (i.e. `getTimeToLiveMs() != null`)  we should use is instead of the value of `org.opennms.jms.timeout` property.